cassandra-mesos-model/src/main/java/io/mesosphere/mesos/util/CassandraFrameworkProtosUtils.java

```diff
@@ -172,6 +180,83 @@ public static CassandraNodeTask getTaskForNode(@NotNull final CassandraNode cass
         return builder;
     }
 
+    @NotNull
+    public static Function<Resource, TreeSet<Long>> resourceToPortSet() {
+        return new Function<Resource, TreeSet<Long>>() {
+            @Override
+            @NotNull
+            public TreeSet<Long> apply(@Nullable Resource resource) {
+                return resourceValueRange(Optional.fromNullable(resource));
+            }
+        };
+    }
+
+    public static Predicate<Resource> containsPorts(final Iterable<Long> ports) {
+        return new Predicate<Resource>() {
+            @Override
+            public boolean apply(Resource resource) {
+                TreeSet<Long> portsInResource = resourceValueRange(Optional.fromNullable(resource));
+                return portsInResource.containsAll(newArrayList(ports));
+            }
+        };
+    }
+
+    public static ImmutableListMultimap<String, Resource> resourcesForRoleAndOffer(String role, Protos.Offer offer) {
```

Comment:
Please mark `resource` and `offer` parameters as final and annotate with `@NotNull`.